gis/JOSM/Portfile

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+name                JOSM
+version             16239
+categories          gis editors java
+platforms           darwin
+license             GPL-2+
+
+maintainers         {@sikmir gmail.com:sikmir} openmaintainer
+
+description         An extensible editor for OpenStreetMap
+long_description    ${name} is a feature-rich editor for the \
+                    experienced OSM mappers.
+homepage            https://josm.openstreetmap.de
+
+master_sites        ${homepage}/download/macosx/
+distfiles           josm-macosx.zip
```

Comment:
This is an [unversioned distfile](https://trac.macports.org/wiki/PortfileRecipes#unversioned-distfiles) which is discouraged. Fortunately the project also publishes versioned distfiles, so you should use those.

You also don't need to set `distfiles`; you can set `distname` instead and let `use_zip yes` which you're already setting take care of setting `distfiles` correctly.

So I would change this line to `distname josm-macosx-${version}`.

Reply:
Fixed